aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -848,6 +864,117 @@ impl AptosVM {
         Ok(())
     }
 
+    fn validate_and_execute_batched_calls(
+        &self,
+        resolver: &impl AptosMoveResolver,
+        session: &mut SessionExt,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext,
+        senders: Vec<AccountAddress>,
+        batched_calls: &[BatchedFunctionCall],
+        txn_data: &TransactionMetadata,
+    ) -> Result<(), VMStatus> {
+        let mut batched_return_values = BatchedExecutionResult {
+            return_values: vec![],
+        };
+        for batched_call in batched_calls {
+            self.validate_and_execute_batched_call(
+                resolver,
+                session,
+                gas_meter,
+                traversal_context,
+                senders.clone(),
+                batched_call,
+                &mut batched_return_values,
+                txn_data,
+            )?;
+        }
+        for return_values in batched_return_values.return_values {
```

Comment:
Do we have any limits on # of batched calls? Number of return values?

Reply:
We need limits on the # of calls indeed. We also need a feature flag in the prod version of the PR :D